app/models/research/experiment_solution.rb

```diff
@@ -43,5 +43,13 @@ def latest_files
 
       files
     end
+
+    def submit!(time: Time.current)
```

Comment:
We might want to guard against submitting solutions that haven't passed the tests.